src/usecases/announcement.usecase.js

```diff
@@ -0,0 +1,66 @@
+const Announcement = require("../models/announcement.model")
+const User = require("../models/user.model")
+const School = require("../models/school.model")
+const createError = require('http-errors')
+
+
+// Usecase 1 - getAll
+// con populate
+function getAll () {
+    console.log("imprimiendo desde anuncios, usecase de getAll-populate")
+
+    return Announcement.find({}).populate('user').populate('school')
+}
+
+
+// Usecase 2 - GetById
+// con populate
+const getById = async (id) => {
+    console.log("imprimiendo desde anuncios, usecase dentro de getById")
+
+    const announcement = await Announcement.findById(id).populate('school')
+
+    if(!announcement){
+        const error = createError(404, "anuncio no encontrado")
+        throw error
+    }
+    
+    return announcement
+}
+
+
+// Usecase 3 - Post
+// con populate
+const create = async(newAnnouncement) => {
+    console.log("imprimiendo desde anuncios, usecase dentro de Post")
+
+    // const userFound = await User.findById(userCurrent)
+
+    // if(!userFound) {
+    //     const error = createError(404, "El usuario no fue encontrado")
+    //     throw error
+    // }
+
+    const announcementToCreate = await Announcement.create({newAnnouncement})
```

Comment:
El problema aquí son las llaves, si tu pones llaves aqui, estas crean un objeto con una propiedad `newAnnouncement` eso no es lo que quieres

Quitale las llaves
```suggestion
    const announcementToCreate = await Announcement.create(newAnnouncement)
```

Reply:
Listo, quité las llaves y funcionó.
Muchas gracias.